p2p/antiflood/p2pAntiflood.go

```diff
@@ -50,6 +54,24 @@ func (af *p2pAntiflood) CanProcessMessage(message p2p.MessageP2P, fromConnectedP
 	return nil
 }
 
+// CanProcessMessageOnTopic signals if a p2p message can be processed or not for a given topic
+func (af *p2pAntiflood) CanProcessMessageOnTopic(message p2p.MessageP2P, fromConnectedPeer p2p.PeerID, topic string) error {
+	topicFloodPreventer := af.TopicFloodPreventer
+	if check.IfNil(topicFloodPreventer) {
+		return p2p.ErrNilTopicFloodPreventer
```

Comment:
this is already checked on the constructor

Reply:
also in `CanProcessMessage` there is a nil check. Should I remove that check also ?

Reply:
Right, since we are checking on the underlying composing pointers. Should remain like this. Also, might add a supplemental check on IsInterfaceNil function

Reply:
done